sdl_android_lib/src/com/smartdevicelink/encoder/SdlEncoder.java

```diff
@@ -0,0 +1,159 @@
+package com.smartdevicelink.encoder;
+
+import java.io.IOException;
+import java.io.PipedOutputStream;
+import java.nio.ByteBuffer;
+
+import android.annotation.TargetApi;
+import android.media.MediaCodec;
+import android.media.MediaCodecInfo;
+import android.media.MediaFormat;
+import android.os.Build;
+import android.view.Surface;
+
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+public class SdlEncoder {
+	
+	// parameters for the encoder
+	private static final String _MIME_TYPE = "video/avc"; // H.264/AVC video
+	// private static final String MIME_TYPE = "video/mp4v-es"; //MPEG4 video
+	private static int _FRAME_RATE = 30;
+	private static int _IFRAME_INTERVAL = 5;
+	private static int _FRAME_WIDTH = 800;
+	private static int _FRAME_HEIGHT = 480;
+	private static int _BITRATE = 6000000;
+
+	// encoder state
+	private MediaCodec mEncoder;
+	private PipedOutputStream mOutputStream;
+	
+	// allocate one of these up front so we don't need to do it every time
+	private MediaCodec.BufferInfo mBufferInfo;
+	
+	public SdlEncoder (int frameRate, int iFrameInterval, int width,
+		int height, int bitrate, PipedOutputStream outputStream) {
+		_FRAME_RATE = frameRate;
```

Comment:
There should be "feedback" or ranges supported for each of the variables such as frameRate, bitrate